src/Shared/Recruit.Vacancies.Client/Application/Commands/TransferVacancyToLegalEntityCommand.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Esfa.Recruit.Vacancies.Client.Domain.Messaging;
+using MediatR;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.Commands
+{
+    public class TransferVacancyToLegalEntityCommand : ICommand, IRequest
+    {
+        public long VacancyReference { get; }
+        public Guid UserRef { get; }
+        public string UserEmailAddress { get; }
+        public string UserName { get; }
```

Comment:
I think the TransferReason should be set on the message rather than inferred later on by using the blocked provider lookup.

Reply:
No, the command handler should work this out. If we are processing 100 vacancies individually through this handler and half way through the provider has been blocked, then the rest would get the correct closure reason. This is part of the catchup that our code needs to handle.

Reply:
I disagree and this could bite us.  I think the reason needs to be set at the start because it is the actual event that started the process.

So if we block a provider and kick off a bunch of transfers then quickly remove the block the remaining transfers will be set as 'employer revoked permission'. That is not right.

Reply:
Didn't consider that at all. You are correct. I was trying to keep the handler doing all the necessary lookup type stuff so it is always looking at latest state. However as you mention, in this situation it is not what we would want. I will change so the transfer reason is passed down as you have suggested. I think that way as you said will make the command more explicit about what it is intending to do.